### PR TITLE
Refactor how to record feast metrics.

### PR DESCRIPTION
**Pull request by ariefrahmansyah**
**Which issue(s) this PR fixes**:

Fixes not able to log NULL feature status if user didn't specify default value.

**Checklist**

- [ ] Added unit test, integration, and/or e2e tests
- [ ] Tested locally
- [ ] Updated documentation
- [ ] Update Swagger spec if the PR introduce API changes
- [ ] Regenerated Golang and Python client if the PR introduce API changes

### CODE REVIEW

File: api/pkg/transformer/feast/call.go [hunk 1]
```diff
@@ -82,6 +82,11 @@ func (fc *call) processResponse(feastResponse *feast.OnlineFeaturesResponse) (*i
 			var rawValue *types.Value
 
 			featureStatus := responseStatus[rowIdx][column]
+			// put behind feature toggle since it will generate high cardinality metrics
+			if fc.statusMonitoringEnabled {
```
Comment (pradithya) [inline api/pkg/transformer/feast/call.go:86]: I don't understand why putting it here helps?

Comment (ariefrahmansyah) [reply api/pkg/transformer/feast/call.go:86]: These lines will skip recordMetrics() function: https://github.com/gojek/merlin/blob/5d786310fc9062781ed56710f8570f1e67dfd248/api/pkg/transformer/feast/call.go#L100-L103

So currently, if feature X is NULL, NOT FOUND, or etc, standard transformer tries to find the default value. But if default value is not exist, we will skip it entirely and return it as is without recording the metrics.

Comment (pradithya) [reply api/pkg/transformer/feast/call.go:86]: Can we refactor that part instead? So it shouldn't exit the loop. `processResponse` is already a complex method, I was trying to reduce the noise by extracting metrics recording to `fc.recordMetrics(val, column, featureStatus)`

**Comment by pradithya** [comment]
@ariefrahmansyah Are you going to continue this PR?

**Pull request closed without merging**